workers/googlePayments.js

```diff
@@ -0,0 +1,168 @@
+const nconf = require('nconf');
+const moment = require('moment');
+const request = require('request');
+const iap = require('in-app-purchase');
+const subscriptions = require('../libs/subscriptions');
+const USERS_BATCH = 10;
+// Defined later
+let db;
+let queue;
+let habitrpgUsers;
+
+iap.config({
+  // This is the path to the directory containing iap-sanbox/iap-live files
+  googlePublicKeyPath: nconf.get('IAP_GOOGLE_KEYDIR'),
+  googleAccToken: nconf.get('PLAY_API_ACCESS_TOKEN'),
+  googleRefToken: nconf.get('PLAY_API_REFRESH_TOKEN'),
+  googleClientID: nconf.get('PLAY_API_CLIENT_ID'),
+  googleClientSecret: nconf.get('PLAY_API_CLIENT_SECRET'),
+});
+
+function processUser (user, jobStartDate, nextScheduledCheck) {
+  let plan = subscriptions.blocks[user.purchased.plan.planId];
+
+  if (!plan) {
+    throw new Error(`Plan ${user.purchased.plan.planId} does not exists. User \{user._id}`);
+  }
+  return new Promise((resolve, reject) => {
+    iap.validate(iap.GOOGLE, user.purchased.plan.additionalData, (error, response) => {
+      if (error) {
+        return reject(error);
+      }
+      if (iap.isValidated(response)) {
+        let purchaseDataList = iap.getPurchaseData(response);
+        let subscription = purchaseDataList[0];
+        if (subscription.expirationDate < jobStartDate) {
+          request({
+            url: `${nconf.get('BASE_URL')}/iap/android/subscribe/cancel`,
+            method: 'GET',
+            qs: {
+              noRedirect: 'true',
+              _id: user._id,
+              apiToken: user.apiToken,
+            },
+          }, (habitError, habitResponse, body) => {
+            if (!habitError && habitResponse.statusCode === 200) {
+              return reject(habitError);
```

Comment:
i see, this should be resolve(), in the amazon worker it uses cb() for both success and error but with promises you cannot do it